Add secp256k1 specs (#12)
Nerolation/EIP-ERC-721-Stealth-Addresses: eip-5564.md

}
```

### Stealth meta-address format

The new address format for the stealth meta-addresses is based on [ERC-3770](./eip-3770.md) and extends it by adding a `st:` (*stealth*) as prefix.
Thus, stealth meta-addresses on Ethereum come with the following format:

```
st:eth:0x<spendingKey><viewingKey>
``` 

> *Notably, the address-format is only used to differentiate stealth addresses from standard addresses, however, the prefix is sliced away before doing any computations on the stealth meta-address.*

Reviewer: Nit, think it's a little more clear this way
```suggestion
*Notably, the address-format is only used to differentiate stealth addresses from standard addresses, as the prefix is sliced away before doing any computations on the stealth meta-address.*
```

---

### Initial Implementation of SECP256k1 with View Tags

> This ERC provides a foundation that is not tied to any specific cryptographic system through the IERC5564Messenger contract. In addition, it introduces the first implementation of a stealth address scheme that utilizes the SECP256k1 elliptic curve and`view tags`. The SECP256k1 elliptic curve is defined with the equation $y^2 = x^3 + 7 \pmod{p}$, where $p = 2^{256} - 2^{32} - 977$.

Reviewer: ```suggestion
This ERC provides a foundation that is not tied to any specific cryptographic system through the IERC5564Messenger contract. In addition, it introduces the first implementation of a stealth address scheme that utilizes the SECP256k1 elliptic curve and `view tags`. The SECP256k1 elliptic curve is defined with the equation $y^2 = x^3 + 7 \pmod{p}$, where $p = 2^{256} - 2^{32} - 977$.
```

> The following reference is divided into a stealth address generation, parsing and stealth private key derivation section and is defined as follows:

Reviewer: ```suggestion
The following reference is divided into three sections:
1. Stealth address generation
2. Parsing announcements
3. Stealth private key derivation

 Definitions:
```

Reviewer: We should also link to that page that gives secp256k1 the initial scheme ID

- $G$ represents the generator point of the curve.

Generation - Generate stealth address from stealth meta-address:

> - Recipient has has access to the private keys $p_{spend}$, $p_{view}$ from which $P_{spend}$ and $P_{view}$ are derived.

Reviewer: ```suggestion
- Recipient has has access to the private keys $p_{spend}$, $p_{view}$ from which public keys $P_{spend}$ and $P_{view}$ are derived.
```

- The  `checkStealthAddress` function performs the following computations:
  - Shared secret $s$ is computed by multiplying the viewing private key with the ephemeral public key of the announcement $s = p_{view}$ * $P_{ephemeral}$.
  - The secret is hashed $s_{h} = h(s)$.

>   - The view tag $v$ is extracted by taking the most significant byte $s_{h}[0]$ and can be compared to the given view tag.

Reviewer: ```suggestion
  - The view tag $v$ is extracted by taking the most significant byte $s_{h}[0]$ and can be compared to the given view tag. If the view tags do not match, this `Announcement` is not for the user. If the view tags match, continue on.
```

  - Multiply the hashed shared secret with the generator point $S_h = s_h \cdot G$.
  - The stealth public key is computed as $P_{stealth} = P_{spend} + S_h$.
  - The derived stealth address $a_{stealth}$ is computed as $\textrm{pubkeyToAddress}(P_{stealth})$.
  - Return `true` if the stealth address of the announcement matches the derived stealth address, else return `false`.

Private key derivation - Generate the stealth address private key from the hashed shared secret and the spending private key.

> - User has access to the viewing private key, spending private key $p_{spend}$.

Reviewer: ```suggestion
- User has access to the viewing private key $p_{view}$, spending private key $p_{spend}$.
```

> 

Reviewer: Github won't let me comment on this line, but a few lines below it says:

```
- The stealth private key is computed as $P_{stealth} = p_{spend} + s_h$.
```

but it should say:

```
- The stealth private key is computed as $p_{stealth} = p_{spend} + s_h$.
```

i.e. change the capital `P` to a lowercase `p` since it's a private key
